Sources/MongoSwift/MongoClient.swift

```diff
@@ -53,6 +53,15 @@ public struct ClientOptions: CodingStrategyProvider, Decodable {
      */
     public var serverMonitoring: Bool = false
 
+    /**
+     * `MongoSwift.MongoClient` provides an asynchronous API by running all blocking operations off of their
+     * originating threads in a thread pool. `MongoSwiftSync.MongoClient` is implemented as a wrapper of the async
+     * client which waits for each corresponding asynchronous operation to complete and then returns the result.
+     * This option specifies the size of the thread pool used by the asynchronous client, and determines the max
+     * number of concurrent operations that may be performed using a single client.
+     */
+    public var threadPoolSize: Int? = MongoClient.defaultThreadPoolSize
```

Comment:
One thing to consider when making this configurable is that if we ever remove it, it will be a breaking change. I'm pretty sure we were saying that version 2 would be pure swift (i.e. when this change would happen), but I don't know if we've settled on that for sure.

Reply:
yeah, I thought about that. I think that should indeed be version 2.0. but worst case if it's not for some reason, we can deprecate this and add a compiler warning that it no longer has any effect